inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/repositories/LocalDisplayedMessageRepository.kt

```diff
@@ -79,6 +90,32 @@ internal interface LocalDisplayedMessageRepository {
         override fun clearMessages() {
             if (messages.isNotEmpty()) {
                 messages.clear()
+                saveUpdatedMap()
+            }
+        }
+
+        private fun checkAndResetMap(onLaunch: Boolean = false) {
+            // check if caching is enabled and if there are changes in user info
+            if (InAppMessaging.instance().isLocalCachingEnabled() &&
+                    (onLaunch || user != AccountRepository.instance().userInfoHash)) {
+                user = AccountRepository.instance().userInfoHash
+                // reset message list from cached using updated user info
+                val listString = InAppMessaging.instance().getEncryptedSharedPref()?.getString(LOCAL_DISPLAYED_KEY, "")
+                        ?: ""
+                messages.clear()
+                if (listString.isNotEmpty()) {
+                    val type = object : TypeToken<HashMap<String, List<Long>>>() {}.type
+                    messages.putAll(Gson().fromJson(listString, type))
+                }
+            }
+        }
+
+        private fun saveUpdatedMap() {
+            // check if caching is enabled to update persistent data
+            if (InAppMessaging.instance().isLocalCachingEnabled()) {
+                // reset message list from cached using updated user info
+                InAppMessaging.instance().getEncryptedSharedPref()?.edit()?.putString(LOCAL_DISPLAYED_KEY,
```

Comment:
Can this line fail at any point? If yes, then is it worth it to add error handling even for debug purposes

Reply:
This is not expected to fail, but added debug logs.